.circleci/config.yml

```diff
@@ -61,8 +67,16 @@ jobs:
 
 workflows:
   version: 2
-  build_and_test:
+  checks:
     jobs:
       - check_go_mod
       - check_readme_sync
-      - build
+      - build_and_test
+      - build_and_publish:
+          requires:
+            - build_and_test
+          filters:
+            branches:
+              only: master
```

Comment:
I was reading about the filters earlier and came across this post:  https://discuss.circleci.com/t/dont-run-deploy-workflow-step-for-forked-pr/23606
It suggests that a PR from the master branch of a fork would pass this filter. We might want to consider expanding the check in the publish script to check for one of the environment variables that are only set for forked PRs: https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables

What do you think?

Reply:
Great question.

So you're right on the approach (using env vars) and we do this though by checking the CIRCLE_BRANCH. For pull requests it is `pull/945` or whatnot.

However, as I was typing my response I realized that for the _tagged_ portion of the bash script we could hit a problem. E.g. If I make a PR that includes a tag on my commit I may be able to trigger that part of the build.

We can make this check a bit more verbose by checking for another env var value about the PR, but we should acknowledge that a malicious user could still manipulate the scripts/env vars to trigger these. I dont know if we can prevent that.

Reply:
I just checked the `CIRCLE_BRANCH` in a test PR and you're right, it sets the branch correctly to be a pull request branch:
```
#!/bin/bash -eo pipefail
echo $CIRCLE_BRANCH
echo $CIRCLE_PR_NUMBER
echo $CIRCLE_PR_REPONAME
echo $CIRCLE_PR_USERNAME
pull/946
946
sonobuoy
zubron
```

I'm not sure if there is anything that we can do to prevent malicious PRs. But, given that we've prevented the secrets being passed to forked PRs, they could bypass the checks but the steps shouldn't work because the secrets aren't available.

Reply:
Yes, excellent point. The config settings in the UI prevent image pushing from ever actually occurring.